Bookish/Controllers/MemberController.cs

```diff
@@ -28,49 +29,100 @@ public IActionResult AddMember()
         return View();
     }
     
+    public IActionResult MemberCatalogue()
+    {
+        return View(_dbContext.Members.ToList());
+    }
+    
     
-    public async Task<ActionResult> MemberQuery(Member member)
+    
+    public IActionResult MemberQuery(Member member)
     {
-        if (_dbContext.Members.Find(member.MemberId) == null)
+        if (member.MemberId != null)
+        {
+            return SearchMemberById(member.MemberId);
+        }
+        else if (member.Name != null)
+        {
+            return SearchMemberByName(member.Name);
+        }
+        else if (member.Email != null)
+        {
+            return SearchMemberByEmail(member.Email);
+        }
+        else
+        {
+            return View("ErrorMsg", new ErrorMsgModel("You did not input anything! :("));
+        }
+    }
+
+    private IActionResult SearchMemberById(string memberId)
+    {
+        var memberModel = _dbContext.Members.SingleOrDefault(m => m.MemberId == memberId);
+        if (memberModel == null)
         {
             return View("ErrorMsg", new ErrorMsgModel("Member not found."));
         }
 
         var resultMember = _dbContext.Members
-            .Where(x => x.MemberId == member.MemberId)
+            .Where(x => x.MemberId == memberId)
             .Include(m=>m.Books)
             .Include("Books.Book")
             .ToList()[0];
```

Comment:
I think I already commented about this on Blanca's PR:

```suggestion
var resultMember = _dbContext.Members
    .Where(x => x.MemberId == memberId)
    .Include(m => m.Books)
    .ThenInclude(b => b.Book)
    .SingleOrDefault()
```

Reply:
ok, I have changed it.